utils/loggers/comet/utils.py

```diff
@@ -0,0 +1,127 @@
+import os
+
+import comet_ml
+import yaml
+
+COMET_PREFIX = "comet://"
+COMET_MODEL_NAME = os.getenv("COMET_MODEL_NAME", "yolov5")
+
+
+def download_model_checkpoint(opt, experiment):
+    model_dir = f"{opt.project}/{experiment.name}"
+    os.makedirs(model_dir, exist_ok=True)
+
+    model_name = opt.comet_model_name if opt.comet_model_name else COMET_MODEL_NAME
+    model_asset_list = sorted(
+        experiment.get_model_asset_list(model_name),
+        key=lambda x: x["step"],
+        reverse=True,
+    )
+    checkpoint_step = (
+        opt.comet_checkpoint_step
+        if opt.comet_checkpoint_step
+        else model_asset_list[0]["step"]
+    )
+    checkpoint_filename = opt.comet_checkpoint_filename
+    for asset in model_asset_list:
+        if (int(checkpoint_step) == asset["step"]) and (
+            checkpoint_filename == asset["fileName"]
+        ):
+            asset_id = asset["assetId"]
+            asset_filename = asset["fileName"]
+            break
+
+    try:
+        model_binary = experiment.get_asset(
+            asset_id, return_type="binary", stream=False
```

Comment:
How big would the model checkpoint will be? With `stream=False` I think it will have to put the file content in memory

Reply:
Good catch. The weights could go up to a GB.